clash-lib/src/Clash/Core/TermLiteral/TH.hs

```diff
@@ -21,11 +28,120 @@ import           Clash.Core.Name                 (nameOcc)
 -- module Clash.Core.Subst cannot be linked; it is only available as a boot module
 import Clash.Core.Subst ()
 
+#if __GLASGOW_HASKELL__ >= 900
+type CompatTyVarBndr = TyVarBndr ()
+#else
+type CompatTyVarBndr = TyVarBndr
+#endif
+
 dcName' :: DataCon -> String
 dcName' = Text.unpack . nameOcc . dcName
 
 termToDataName :: Name
-termToDataName = mkName "Clash.Core.TermLiteral.termToData"
+termToDataName =
+  -- Note that we can't use a fully qualified name here: GHC disallows fully
+  -- qualified names in instance function declarations.
+  mkName "termToData"
+
+showsTypePrecName :: Name
+showsTypePrecName =
+  -- Note that we can't use a fully qualified name here: GHC disallows fully
+  -- qualified names in instance function declarations.
+  mkName "showsTypePrec"
+
+termLiteralName :: Name
+termLiteralName = mkName "Clash.Core.TermLiteral.TermLiteral"
+
+-- | Extracts variable names from a 'TyVarBndr', errors if it is not a simply
+-- typed variable name.
+typeVarName :: CompatTyVarBndr -> Q Name
+typeVarName = \case
+#if __GLASGOW_HASKELL__ >= 900
+  PlainTV typVarName () -> pure typVarName
+  KindedTV typVarName () StarT -> pure typVarName
+#else
+  PlainTV typVarName -> pure typVarName
+  KindedTV typVarName StarT -> pure typVarName
+#endif
+  k@(KindedTV {}) -> fail $ "Not supported: KindedTV: " <> show k
+
+-- | Derive a 'TermLiteral' instance for given type.
```

Comment:
```suggestion
-- | Derive a t'Clash.Core.TermLiteral.TermLiteral' instance for given type
```

It's not in scope. Also, it's a sentence fragment if I'm editing it anyway :-)

Reply:
Did you add the `t` on purpose?

Reply:
Answered [here](https://github.com/clash-lang/clash-compiler/pull/2329#discussion_r971732826)